### PR TITLE
Persist the stream before getting finish ACK

### PR DESCRIPTION
**Pull request by cyx**
For some cases, we don't ever persist whatever we get, and
thus makes it hard to debug what the build was doing before
it hung completely.

This should persist the existing output every X minutes (set
to 5 minutes) so that we can get a better understanding of the
failure conditions for certain cases.

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -90,8 +91,24 @@ func pub(w http.ResponseWriter, r *http.Request) {
...
+				return
+			case <-time.After(*util.StorageInterval):
+				// Asynchronously upload the output to our defined storage backend.
+				go storeOutput(key(r), requestURI(r))
+			}
+		}
+
+	}()
+
+	close(done)
```
Comment (tt) [inline server/server.go]: When this is closed, the routine will immediately pick up the first case, so there is really no change here.

**Comment by tt** [comment]
I don't think we should write every X minutes. We can write whenever a publisher closes the connection; that will definitely happen at some point.

**Comment by cyx** [comment]
Closing in favor of #59

**Comment by cyx** [comment]
Leaving this open in case #59 fails to do its job.

**Comment by cyx** [comment]
#59 seems to do its job! Closing

File: server/server.go [hunk 2]
```diff
@@ -90,8 +108,7 @@ func pub(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	// Asynchronously upload the output to our defined storage backend.
-	go storeOutput(key(r), requestURI(r))
+	close(done)
```
Comment (freeformz) [inline server/server.go]: As discussed in hipchat ... You need to defer this above or you'll leak go routines.

Comment (cyx) [inline server/server.go]: Thanks!

**Comment by tt** [comment]
Why was this reopened? Can we change to write on publisher disconnects to reduce the number of writes?

**Comment by cyx** [comment]
Mostly reopened to track the fact that the hanging builds still aren't
getting captured. The problem is that the forever hanging doesn't
disconnect.

We can probably do something there and change busltee to disconnect after X
seconds.

**Comment by tt** [comment]
"Forever" is limited to 24 hours by the nature of running inside a dyno.

**Comment by cyx** [comment]
Yep, but 24 hours extension is probably something too risky. You guys can
play with that value though, but my guess is that it might cause huge jumps
in redis usage.

**Comment by dmathieu** [comment]
Do we still need this?
I'm guessing if we've been able to live without for almost a year, it's not so much of an issue?

**Comment by dmathieu** [comment]
Closed this as stale.

**Pull request closed without merging**